NTuplizers/python/userMETs_cff.py

```diff
@@ -7,77 +7,92 @@ def userMETs(process, isData, era):
     from PhysicsTools.PatAlgos.slimming.puppiForMET_cff import makePuppiesFromMiniAOD
 
     # https://twiki.cern.ch/twiki/bin/view/CMS/MissingETUncertaintyPrescription#Puppi_MET
-    makePuppiesFromMiniAOD(process, True)
-
-    if 'era' == '2016':
+    
+    process.noMuCands = cms.EDFilter("CandPtrSelector",
+                                 src=cms.InputTag("packedPFCandidates"),
+                                 cut=cms.string("abs(pdgId)!=13")
+                                 )
 
-       runMetCorAndUncFromMiniAOD(process, isData = isData,
+    makePuppiesFromMiniAOD(process, True)
+    if era == '2016':
+       runMetCorAndUncFromMiniAOD(process, isData = isData,recoMetFromPFCs=True,
```

Comment:
As far as I understand this is needed to recalculate the pfMet https://github.com/cms-sw/cmssw/blob/master/PhysicsTools/PatUtils/python/tools/runMETCorrectionsAndUncertainties.py#L339 
Otherwise only the corrections are recalculated